0021/README.md

```diff
@@ -0,0 +1,155 @@
+{
+    "namespace":"org.kaaproject.ipc.tmmp.gen.v1",
+    "name":"TenantMetadataGetRequest",
+    "type":"record",
+    "doc":"Tenant metadata get request",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"appName",
+            "type":[
+              "null",
+              "string"
+            ],
+            "default":null,
+            "doc":"Application that tenant information is requested for"
+        },
+        {
+            "name":"appVersionName",
+            "type":[
+              "null",
+              "string"
+            ],
+            "default":null,
+            "doc":"Application version that tenant information is requested for"
+        }
+    ]
+}
+```
+
+Note that `appName` and `appVersionName` fields are declared as optional. 
+A client MUST specify at least one of these fields so that a repository could resolve which application tenant metadata is requested for.
+
+
+## Tenant metadata get response
+
+*Tenant metadata get response* message MUST be sent by repository in response to an [tenant metadata get request](#tenant-metadata-get-request).
+Repository MUST publish tenant metadata get response message to the subject provided in the NATS `replyTo` field of the request.
+
+Tenant metadata get response message payload MUST be an Avro-encoded object with the following schema ([0021-tenant-metadata-get-response.avsc](./0021-tenant-metadata-get-response.avsc)):
+
+```json
+{
+    "namespace":"org.kaaproject.ipc.tmmp.gen.v1",
+    "name":"TenantMetadataGetResponse",
+    "type":"record",
+    "doc":"Tenant metadata get response",
+    "fields":[
+        {
+            "name":"correlationId",
+            "type":"string",
+            "doc":"Message ID primarily used to track message processing across services"
+        },
+        {
+            "name":"timestamp",
+            "type":"long",
+            "doc":"Message creation UNIX timestamp in milliseconds"
+        },
+        {
+            "name":"timeout",
+            "type":"long",
+            "default":0,
+            "doc":"Amount of milliseconds (since the timestamp) until the message expires. Value of 0 is reserved to indicate no expiration."
+        },
+        {
+            "name":"tenantId",
+            "type":"string",
+            "doc":"Tenant ID"
+        },
+        {
+            "name":"appName",
+            "type":"string",
+            "doc":"Tenant's application"
```

Comment:
What is this? How does it correlate with app / appV in the request? Why is the client interested in this?